R/dist_quantiles.R

```diff
@@ -5,6 +5,9 @@ new_quantiles <- function(values = double(), quantile_levels = double()) {
   vec_cast(values, double())
   vec_cast(quantile_levels, double())
   stopifnot(length(values) == length(quantile_levels))
+  if (length(values) == 0L) {
+    return(distributional::dist_degenerate(double()))
```

Comment:
issue: this is the wrong type to come out of `new_quantiles()`; `dist_degenerate(double())` is a `"distribution"`, not an element thereof.  Plus it seems like asking for trouble anyway to have these ill-formed distributions around.

```suggestion
    cli_abort("new_quantiles() requires at least one quantile to be specified.  If you meant to make an empty list of distributions, try dist_quantiles() instead.")
```

Reply:
agreed. Also changing the function defaults to `double(1)` to avoid triggering by default.